app/vmalert/notifier/alert.go

```diff
@@ -21,6 +21,8 @@ type Alert struct {
 	GroupID uint64
 	// Name represents Alert name
 	Name string
+	// Type defines the datasource type of the Alert
+	Type string
```

Comment:
`Type` is a bit too broad and ambiguous. Naming it  `Datasource` would make its purpose clearer. WDYT?

Reply:
Type here goes from:
> Optional type for expressions inside rules to override the `-rule.defaultRuleType(default is "prometheus")` cmd-line flag.
Supported values: "graphite", "prometheus" and "vlogs"(check https://docs.victoriametrics.com/victorialogs/vmalert/ for details).
[ type: <string> ]

in https://docs.victoriametrics.com/victoriametrics/vmalert/#groups

Do you think renaming it into `Datasource` won't result into confusion?

Reply:
`Datasource` + `Type` = `DatasourceType`

Reply:
I don't rly like the `DatasourceType` reference, because it is not about datasource. It is an expression type, as stated in Group config. The `-datasource.url` in vmalert has no type.

Reply:
I'll keep it as is, because `type` is already used as is in alerting and recording rules properties. As well as exposed via APIs - see https://play.victoriametrics.com/select/0/prometheus/vmalert/api/v1/rules